salt/common/tools/sbin/so-status

```diff
@@ -290,22 +291,29 @@ main() {
 
 # {% endraw %}
 
+is_tty() {
+    __tty=0
+    [ -t 1 ] \
+        && __tty=1
+    # don't print colors if NO_COLOR is set to anything
+    [ "${#NO_COLOR}" -ne 0 ] \
+        && __tty=0
+    return "${__tty}"
+}
+
+if ! [ "$(id -u)" = 0 ]; then
+   echo "${MYNAME}: This command must be run as root"
+   display_help 1
+fi
```

Comment:
This is included at the top of the script so nothing else is evaluated before checking for root, and displaying the usage message here also breaks from the standard set by other scripts in Security Onion.

Reply:
But what if all I want to do when running is to check the script's usage? Should you really need root just to see the usage?

There's probably a better way to handle that use case, while staying consistent with the rest of the scripts.

Reply:
I think there's a valid argument to be made for parsing arguments before checking root and showing usage if that flag is passed in, but any change like that should be made to all Security Onion scripts to maintain consistency.

So for now, let's revert this change. But feel free to [start a discussion](https://github.com/Security-Onion-Solutions/securityonion/discussions/new) to put forward the idea of handling this differently and the team can look it over.

Reply:
Does this scrip even need to run as root?

One nice consequence of using the data from the socket is that the calling user doesn't even need to be in the docker group.

Reply:
To test this script as a regular user I removed the check completely.

Reply:
If you notice, these scripts are all included in `/usr/sbin`, so by convention (and by design) they require root.

Reply:
Yea I'll probably start a discussion about it. That whole principal of least privilege and everything.

Reply:
Discussion made https://github.com/Security-Onion-Solutions/securityonion/discussions/7156